src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.basic
+
+/-!
+# Lexicographic order on finsupps
+
+This file defines the lexicographic order on finsupps.
+-/
+
+lemma ne.or_ne_of_ne {α} {a b : α} (c : α) (ab : a ≠ b) :
```

Comment:
Sorry, what does this have to do with "exists"? I agree that it doesn't belong to this file, as it's basically just the contrapositive of `eq.trans`:
```
import logic.basic
lemma ne.or_ne_of_ne {α} {a b : α} (c : α) (ab : a ≠ b) : a ≠ c ∨ b ≠ c := 
not_and_distrib.1 $ mt (and_imp.2 eq.substr) ab.symm
```

Reply:
`exists_ne` is an immediate consequence of this lemma applied to two different elements in a nontrivial type.

Reply:
Oh, I see!

Reply:
Thank you both!  I had some reservations about this lemma and the only reason I stated it, was that I could not find a proof that fit on a line.  I inlined Junyan Xu's proof and removed the lemma.

Reply:
PR #16023 contains this lemma: going forwards, it seems that I am going to use it a couple more times.

I did not use it to reduce `exists_ne`, since what I tried introduced axioms and I imagine that the `decidable` lemmas try to avoid this.